src/server/main_service.cc

```diff
@@ -1469,6 +1469,10 @@ void Service::Publish(CmdArgList args, ConnectionContext* cntx) {
   int num_published = subscribers.size();
 
   if (!subscribers.empty()) {
+    // Make sure neither of the subscribers buffers is filled up.
```

Comment:
maybe add here that it's not 100% fail-proof since several publishers can proceed at the same time after getting a green light.

Reply:
Its not fool proof even with one 🙂